electrum_nmc/electrum/names.py

```diff
@@ -132,6 +132,62 @@ def validate_value_length(value: bytes):
     if value_length > value_length_limit:
         raise BitcoinException('value length {} exceeds limit of {}'.format(value_length, value_length_limit))
 
+def validate_onion_address(address: str) -> None:
+    # Verify and remove ".onion" from the address if present
+    if not address.endswith(".onion"):
+        raise ValueError("Invalid onion address: missing .onion suffix")
+
+    # Remove ".onion" from the address
+    address = address[:-len('.onion')]  # Slice slicing to remove the last 6 characters (length of ".onion")
+
+    try:
+        # Initialize the bytearray to hold the decoded service ID
+        decoded_service_id = bytearray([0] * V3_ONION_SERVICE_ID_RAW_SIZE)
+
+        # Decode the service ID from base32 into the decoded_service_id bytearray
+        decoded_service_id = base64.b32decode(address.encode(), decoded_service_id)
```

Comment:
The 2nd argument (per docs) is `casefold`, which accepts a boolean. So this looks wrong. Also we should enforce that the .onion domain is fully lowercase.

Reply:
We'll need to set the text to uppercase, passing it as lowercase doesn't work